homework.js

```diff
@@ -0,0 +1,170 @@
+// Summation 
+
+const summation = function (n){ 
+    let num=0;
+    for (i=1; i<=n ;i++)
+{
+     num=num+i;
+
+}
+console.log(num)
+};
+
+//
+
+const sumEven = function (n){ 
+    let num=0;
+    for (i=1; i<=n ;i++)
+{
+    if (i%2 === 0)
+     num=num+i;
+
+}
+console.log(num)
+};
+
+//
+
+const no=[]
+no.push(8,2,2,4);
+length= no.length-1 ;
+let num=0;
+for ( i=0 ; i<= length ; i++ )
+{ 
+    num=num+no[i];
+}
+console.log(num/no.length)
+
+
+
+
+
+let reverse= function (str) {
+// first split the litters of the word or string , then reverse , then joing the elements. 
+let splitStr = str.split("");
+let reArray = splitStr.reverse();
+let joinArray = reArray.join("");
+return joinArray;
+}
+reverse('elo');
+reverse("hello");
+
+
+
+
+const addDashes = function ()
+{
+   const testArray=['test1', 'test2', 'test3'];
+    for ( let i=0 ; i<testArray.length ; i++)
+    {
+        console.log(testArray[i]+" - ");
+    }
+}
+
+addDashes();
+
+
+const Count = function (n)
+{ 
+    for ( let i=1 ; i<=n ; i++)
+    { 
+        console.log(i); 
+
+    }; 
+        n= n-1; 
+
+    while ( n > 0)
+    { 
+        console.log(n);
+        n--;
+    }
+}
+
+
+
+
+/*
+<li>Write a function that will tell you all of the words in an array that contain a specified letter
+  <br>
+    <code > wordsWithLetter("g", ['cat', 'rabbit', 'dog', 'frog']) // should return ['dog', 'frog']</code>
+  </li> 
+  */
+
+
+
+const wordsWithA = function ()
```

Comment:
youre almost done with this code. we want to get back an array or a log of only the words with the specified letter

for example when we call the function like this we get only words with a g

wordsWithLetter("g", ['cat', 'rabbit', 'dog', 'frog']) // should return ['dog', 'frog']